error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/MockitoMockClassReference.java

```diff
@@ -45,11 +51,13 @@ public final class MockitoMockClassReference extends BugChecker
   private static final Matcher<MethodInvocationTree> MOCKITO_MOCK_OR_SPY =
       allOf(
           argumentCount(1),
-          argument(0, isSameType(Class.class.getName())),
+          argument(0, allOf(isSameType(Class.class.getName()), not(isVariable()))),
```

Comment:
There can be caveats when the passed argument is a variable that can lead into non-compilable/working-with-Mockito code, e.g. the following is odd and won't work:

```diff
-<T> T getMock(class<T> clazz) {
-  return mock(clazz);
+<T> T getMock() {
+  return mock();
}


@Test
void mockType() {
-  assertThat(getMock(BigInteger.class)).isInstanceOf(BigInteger.class);
+  assertThat((BigInteger) getMock()).isInstanceOf(BigInteger.class);
}
```

There might be cases where we could still replace the variable, but I couldn't find an elegant way for it without going into the caveats. For example the trivial:

```diff
-Class<BigInteger> clazz = BigInteger.class;
-BigInteger mock = mock(clazz);
+BigInteger mock = mock();
}
```

Feel free to play around and see if you find an elegant way.